homeassistant/components/econet/binary_sensor.py

```diff
@@ -4,40 +4,51 @@
 from pyeconet.equipment import EquipmentType
 
 from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_LOCK,
     DEVICE_CLASS_OPENING,
     DEVICE_CLASS_POWER,
+    DEVICE_CLASS_SOUND,
     BinarySensorEntity,
 )
 
 from . import EcoNetEntity
 from .const import DOMAIN, EQUIPMENT
 
-_LOGGER = logging.getLogger(__name__)
-
-SENSOR_NAME_RUNNING = "running"
 SENSOR_NAME_SHUTOFF_VALVE = "shutoff_valve"
-SENSOR_NAME_VACATION = "vacation"
+SENSOR_NAME_RUNNING = "running"
+SENSOR_NAME_SCREEN_LOCKED = "screen_locked"
+SENSOR_NAME_BEEP_ENABLED = "beep_enabled"
+
+SENSOR_NAMES_TO_ATTRIBUTES = {
+    SENSOR_NAME_SHUTOFF_VALVE: "shutoff_valve_open",
+    SENSOR_NAME_RUNNING: "running",
+    SENSOR_NAME_SCREEN_LOCKED: "screen_locked",
+    SENSOR_NAME_BEEP_ENABLED: "beep_enabled",
+}
+
+SENSOR_NAMES_TO_DEVICE_CLASS = {
+    SENSOR_NAME_SHUTOFF_VALVE: DEVICE_CLASS_OPENING,
+    SENSOR_NAME_RUNNING: DEVICE_CLASS_POWER,
+    SENSOR_NAME_BEEP_ENABLED: DEVICE_CLASS_SOUND,
+    SENSOR_NAME_SCREEN_LOCKED: DEVICE_CLASS_LOCK,
+}
+
+_LOGGER = logging.getLogger(__name__)
 
 
 async def async_setup_entry(hass, entry, async_add_entities):
     """Set up EcoNet binary sensor based on a config entry."""
     equipment = hass.data[DOMAIN][EQUIPMENT][entry.entry_id]
     binary_sensors = []
-    for water_heater in equipment[EquipmentType.WATER_HEATER]:
-        if water_heater.has_shutoff_valve:
-            binary_sensors.append(
-                EcoNetBinarySensor(
-                    water_heater,
-                    SENSOR_NAME_SHUTOFF_VALVE,
-                )
-            )
-        if water_heater.running is not None:
-            binary_sensors.append(EcoNetBinarySensor(water_heater, SENSOR_NAME_RUNNING))
-        if water_heater.vacation is not None:
-            binary_sensors.append(
-                EcoNetBinarySensor(water_heater, SENSOR_NAME_VACATION)
-            )
-    async_add_entities(binary_sensors)
+    all_equipment = equipment[EquipmentType.WATER_HEATER]
+    all_equipment.extend(equipment[EquipmentType.THERMOSTAT])
+    for _equip in all_equipment:
+        for name, attribute in SENSOR_NAMES_TO_ATTRIBUTES.items():
+            if getattr(_equip, attribute, None) is not None:
+                binary_sensors.append(EcoNetBinarySensor(_equip, name))
+    async_add_entities(
+        binary_sensors,
+    )
```

Comment:
Please, don't combine different things in a single PR. Split this PR up in 2

Reply:
Water heaters and thermostats share some of the same sensors. Trying to reduce the duplicate logic that would be required in order to provide all of the thermostats sensors.